### PR TITLE
Run Labels refactor

### PR DESCRIPTION
**Pull request by cicdw**
## Summary

This PR refactors `labels` to be specified at the flow run level (as opposed to the flow / flow group level).  Moreover, the migration will set the labels on all currently `Scheduled` runs to the correct values so that no work is missed and no backwards compatibility issues arise within the application code.

## Importance
- consolidates all our label logic into one place: `create_flow_run`
- will allow the UI to query for `flow_run{labels}` instead of recreating the flow / flow group logic as it currently does
- will allow users to configure labels on a _per run_ basis
- will allow users to schedule a flow to run multiple times simultaneously with different label sets

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)

### CODE REVIEW

File: src/prefect_server/api/runs.py [hunk 1]
```diff
@@ -155,6 +174,18 @@ async def _create_flow_run(
     elif flow.archived:
         raise ValueError(f"Flow {flow.id} is archived.")
 
+    # set labels
+    run_labels = []
```
Comment (jcrist) [inline src/prefect_server/api/runs.py:178]: This is never hit, the `else` branch handles it.

File: services/postgres/alembic/versions/2020-09-30T123806_add_label_column_to_flow_runs.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+down_revision = "850b76d44332"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column(
+        "flow_run", sa.Column("labels", JSONB, nullable=False, server_default="[]")
+    )
+    op.execute(
```
Comment (jlowin) [inline services/postgres/alembic/versions/2020-09-30T123806_add_label_column_to_flow_runs.py:25]: Just for future reference do you mind adding a small comment here - something like "for runs scheduled at the time this migration is executed, we want to make sure they have the new logic."

**Review by jlowin — CHANGES_REQUESTED** [review]

File: src/prefect_server/api/runs.py (hunk 1 above)
Comment (cicdw) [reply src/prefect_server/api/runs.py:178]: I usually do this for readability so it's clear I expect this to be a list, but I can remove

File: services/postgres/alembic/versions/2020-09-30T123806_add_label_column_to_flow_runs.py (hunk 2 above)
Comment (cicdw) [reply services/postgres/alembic/versions/2020-09-30T123806_add_label_column_to_flow_runs.py:25]: Updated

File: services/postgres/alembic/versions/2020-09-30T123806_add_label_column_to_flow_runs.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+def upgrade():
+    op.add_column(
+        "flow_run", sa.Column("labels", JSONB, nullable=False, server_default="[]")
+    )
+    # this query looks at all currently scheduled flow runs, or more generally any flow
+    # run that has been updated in the last day, and applies our standard label logic to it.
+    # This ensures that active flow runs don't incorrectly receive `[]` as their label set,
+    # which would cause delayed work and user frustration.
+    # Note that flow runs not touched by this query will have `[]` has their labels regardless of
```
Comment (jlowin) [inline services/postgres/alembic/versions/2020-09-30T123806_add_label_column_to_flow_runs.py:29]:
```suggestion
    # Note that flow runs not touched by this query will have `[]` as their labels regardless of
```

**Review by jlowin — APPROVED** [review]

**Pull request merged**